sdk/core/Azure.Core/src/Pipeline/Internal/RetryPolicy.cs

```diff
@@ -127,6 +127,9 @@ private async ValueTask ProcessAsync(HttpMessage message, ReadOnlyMemory<HttpPip
                     }
                 }
 
+                // Dispose the content stream to free up a connection if the request has any
+                message.Response.ContentStream?.Dispose();
```

Comment:
Should this also happen in line 110 and/or 99 when retry loop exists with failure?

Reply:
At that point the stream become the responsibility of the caller. I.e. the logic that creates an exception would read and dispose the stream.

We just need to make sure that no streams are lost in retry process.